apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Query.cs

```diff
@@ -1077,18 +1123,18 @@ internal InternalOrdering(FieldPath field, Direction direction)
             public bool Equals(InternalOrdering other) => Field.Equals(other.Field) && Direction == other.Direction;
         }
 
-        private struct InternalFilter : IEquatable<InternalFilter>
+        internal class InternalFilter : IEquatable<InternalFilter>
```

Comment:
I'd expect to be able to get rid of this entirely - potentially having other subclasses of Filter if necessary.